service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapperTest.java

```diff
@@ -232,12 +247,44 @@ private static Stream<Arguments> resourceFileParams() {
     }
 
     @Test
-    public void When_MappingImmunizationWithInvalidPractitionerReferenceType_Expect_Error() throws IOException {
+    public void When_MappingImmunizationWithInvalidPractitionerReferenceType_Expect_Error() {
         var jsonInput = ResourceTestFileUtils.getFileContent(INPUT_JSON_WITH_PRACTITIONER_INVALID_REFERENCE_RESOURCE_TYPE);
         Immunization parsedImmunization = fhirParseService.parseResource(jsonInput, Immunization.class);
 
         assertThatThrownBy(() -> observationStatementMapper.mapImmunizationToObservationStatement(parsedImmunization, false))
             .isExactlyInstanceOf(EhrMapperException.class)
             .hasMessage("Not supported agent reference: Patient/6D340A1B-BC15-4D4E-93CF-BBCB5B74DF73");
     }
+
+    @Test
+    public void When_ConfidentialityServiceReturnsConfidentialityCode_Expect_MessageContainsConfidentialityCode() {
```

Comment:
Just thinking of test name clarity - maybe something like:

```suggestion
    public void When_MappingImmunization_With_NopatMetaSecurity_Expect_ConfidentialityCodeWithinObservationStatement() {
```

(OPTIONAL)